cvxpy/expressions/leaf.py

```diff
@@ -273,18 +274,73 @@ def is_complex(self) -> bool:
         """
         return self.attributes['complex'] or self.is_imag() or self.attributes['hermitian']
 
+    def _has_lower_bounds(self) -> bool:
+        """Does the variable have lower bounds?"""
+        if self.is_nonneg():
+            return True
+        elif self.attributes['bounds'] is not None:
+            lower_bound = self.attributes['bounds'][0]
+            if np.isscalar(lower_bound):
+                return lower_bound != -np.inf
+            else:
+                return np.any(lower_bound != -np.inf)
```

Comment:
Is there an else case missing here?
```suggestion
            else:
                return np.any(lower_bound != -np.inf)
        else:
            return False
```

Reply:
Added.